### PR TITLE
ProductDetails: log warning when product title is empty

### LINKED ISSUE #624: Add log warning when product.title is empty string, nil, or just filled with spaces.
I think RevenueCat/purchases-flutter#212 could be debugged easier if we added some more logging around product fields that we expect to not be empty.

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes #624.

### CODE REVIEW

File: Purchases/Purchasing/ProductDetails.swift [hunk 1]
```diff
@@ -24,6 +24,14 @@ public typealias SK2Product = StoreKit.Product
 
 /// Abstract class that provides access to all of StoreKit's product type's properties.
 @objc(RCProductDetails) public class ProductDetails: NSObject {
+    public override init() {
+        super.init()
+
+        if self.localizedTitle.isEmpty {
```
Comment (NachoSoto) [inline Purchases/Purchasing/ProductDetails.swift:30]: We can change this to `self.localizedTitle.trimmingWhitespacesAndNewLines.isEmpty` once #995 is merged.

File: Purchases/Purchasing/ProductDetails.swift [hunk 2]
```diff
@@ -106,6 +116,8 @@ public typealias SK2Product = StoreKit.Product
 
     @objc public init(sk1Product: SK1Product) {
         self.underlyingSK1Product = sk1Product
+
+        super.init()
```
Comment (taquitos) [inline Purchases/Purchasing/ProductDetails.swift:120]: good catch with the missing inits on the `NSObject` subclass

File: Purchases/Purchasing/ProductDetails.swift [hunk 3]
```diff
@@ -24,6 +24,14 @@ public typealias SK2Product = StoreKit.Product
 
 /// Abstract class that provides access to all of StoreKit's product type's properties.
 @objc(RCProductDetails) public class ProductDetails: NSObject {
+    public override init() {
+        super.init()
+
+        if self.localizedTitle.isEmpty {
+            Logger.warn("Empty Product titles are not supported: \(self)")
```
Comment (NachoSoto) [inline Purchases/Purchasing/ProductDetails.swift:31]: Oh wait, this needs to go to `Strings`

**Review by taquitos — APPROVED** [review]
🖇️🐐

File: Purchases/Logging/Strings/OfferingStrings.swift [hunk 4]
```diff
@@ -37,6 +37,7 @@ enum OfferingStrings {
     case configuration_error_skproducts_not_found
     case configuration_error_no_products_for_offering
     case offering_empty(offeringIdentifier: String)
+    case product_details_empty_title(productIdentifier: String)
```
Comment (NachoSoto) [inline Purchases/Logging/Strings/OfferingStrings.swift:40]: Not sure if this is the right file but it seemed like the best choice.

**Pull request merged**